client/src/components/home/index.tsx

```diff
@@ -84,6 +85,13 @@ function HomePage(props: {
     startTraining,
   } = useWithReviewAnswerState(props.accessToken, props.search);
 
+  const buildAction = mentor?.isDirty
+    ? startTraining
+    : (mentorId: string) => {
+        const path = `${location.origin}${CLIENT_ENDPOINT}?mentor=${mentorId}`;
```

Comment:
If `startTraining` is an action provided by `useWithReviewAnswerState`, shouldn't `launchMentor` be an action as well?